### PR TITLE
feat(CLI): add fish completion

### PR DESCRIPTION
**Pull request by arugal**
https://github.com/tensorchord/envd/issues/1377

Hi @hezhizhen, I found that `envd bootstrap` does not currently support autocomplete for fish shell, so I submitted a PR for it. If you're also working on implementing this, please let me know.

### CODE REVIEW

File: pkg/autocomplete/fish.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/tensorchord/envd/pkg/util/fileutil"
+)
+
+func InsertFishCompleteEntry(clicontext *cli.Context) error {
+	_, err := exec.LookPath("fish")
+	if err != nil {
+		return errors.Errorf("can't find fish in this system, stop settings the fish-completion")
+	}
+
+	homeDir := os.Getenv("HOME")
```
Comment (kemingy) [inline pkg/autocomplete/fish.go:34]: Can use `os.UserHomeDir` directly.

File: pkg/autocomplete/fish.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	if !dirPathExists {
+		return errors.Errorf("unable to enable fish-completion: %s does not exists", dirPath)
+	}
+
+	pathExists, err := fileutil.FileExists(path)
+	if err != nil {
+		return errors.Wrapf(err, "failed checking if %s exists", path)
+	}
+	if pathExists {
+		// file already exists, don't update it
```
[the remark this answers is not in the document]
Comment (arugal) [reply pkg/autocomplete/fish.go:51]: Okay, I referenced [zsh](https://github.com/tensorchord/envd/blob/main/pkg/autocomplete/zsh.go#L95) and [bash](https://github.com/tensorchord/envd/blob/main/pkg/autocomplete/bash.go#L73)Okay. Do I need to update them together?

[the remark this answers is not in the document]
Comment (kemingy) [reply pkg/autocomplete/fish.go:51]: Yeah. That will be great.

[the remark this answers is not in the document]
Comment (arugal) [reply pkg/autocomplete/fish.go]: Done, please review:)

**Review by kemingy — APPROVED** [review]
Thanks for your contribution :tada:

**Pull request merged**